src/components/adminDashboard/components/eMetadata/controllers/educatorProfile.controller.js

```diff
@@ -0,0 +1,84 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra-web-app.adminDashboard')
+        .controller('EducatorProfileController', function ($mdDialog, $timeout, userProfile, timezonesList, localTimezone, ZnkToastSrv, EMetadataService, $filter) {
+            'ngInject';
+            var self = this;
+            var translateFilter = $filter('translate');
+
+            self.timezonesList = timezonesList;
+            self.profileData = userProfile;
+            self.profileData.educatorTeachworksName = self.profileData.educatorTeachworksName || self.profileData.nickname;
+            self.profileData.timezone = localTimezone;
+            self.profileData.educatorAvailabilityHours = self.profileData.educatorAvailabilityHours || translateFilter("ADMIN.EMETADATA.FROM_TO");
+            self.isTimezoneManual = false;
+
+            self.closeDialog = function () {
+                $mdDialog.cancel();
+            };
+
+            self.updateProfileTimezone = function () {
+                if (!self.profileData.isTimezoneManual) {
+                    self.profileData.timezone = localTimezone;
+                }
+            };
+            self.updateProfile = function (profileform) {
+
+                var type, msg;
+
+                if (profileform.$valid && profileform.$dirty) {
+                    EMetadataService.updateProfile(self.profileData).then(function () {
+                        $timeout(function () {
```

Comment:
to be honest, I reused some code I found, but I don't think there's any added value by using the $timeout, I'll remove it.